recipes-extended/dpdk/dpdk.inc

```diff
@@ -0,0 +1,121 @@
+DESCRIPTION = "Data Plane Development Kit"
+HOMEPAGE = "http://dpdk.org"
+LICENSE = "BSD-3-Clause & LGPLv2 & GPLv2"
+LIC_FILES_CHKSUM = "file://license/README;md5=3383def2d4c82237df281174e981a492"
+
+SRC_URI = "git://source.codeaurora.org/external/qoriq/qoriq-components/dpdk;nobranch=1 \
+           file://add-RTE_KERNELDIR_OUT-to-split-kernel-bu.patch \
+           file://0001-add-Wno-cast-function-type.patch \
+           file://0001-Add-RTE_KERNELDIR_OUT.patch \
+           file://0004-update-WERROR_FLAGS.patch \
+"
+SRCREV = "0123ced10233e7de8a72f743e2ae7e9791124f07"
+
+# Don't use the default value of datadir and let the dpdk set it to a dpdk
+# specific directory
+datadir[unexport] = "1"
+
+RDEPENDS_${PN} += "python-subprocess"
+DEPENDS = "virtual/kernel openssl"
+do_configure[depends] += "virtual/kernel:do_shared_workdir"
+
+inherit module
+
+export MODULE_DIR="/lib/modules/${KERNEL_VERSION}/kernel/drivers/net"
+export RTE_SDK = "${S}"
+export RTE_TARGET = "${ARCH}-dpaa-linuxapp-gcc"
+export ICP_ROOT = "${PKG_CONFIG_SYSROOT_DIR}/usr/include"
+export ICP_LIB_ROOT= "${PKG_CONFIG_SYSROOT_DIR}/usr/lib"
+export RTE_KERNELDIR = "${STAGING_KERNEL_DIR}"
+export RTE_KERNELDIR_OUT = "${STAGING_KERNEL_BUILDDIR}"
+export INSTALL_PATH = "${prefix}/share/dpdk"
+export RTE_OUTPUT = "${S}/${RTE_TARGET}"
+export ETHTOOL_LIB_PATH = "${S}/examples/ethtool/lib/${RTE_TARGET}/"
+export SYSROOTPATH = "--sysroot=${STAGING_DIR_HOST}"
+export ICP_LAC_API_DIR = "${STAGING_DIR_TARGET}${includedir}/lac"
+export OPENSSL_PATH = "${STAGING_DIR_HOST}"
+export EXAMPLES_BUILD_DIR = "${RTE_TARGET}"
```

Comment:
We have standard variables for this so please set this on EXTRA_OEMAKE and in other fields refer to our standard variables such as  `datadir`, and so on.